src/services/account/useAvailablePlans.ts

```diff
@@ -0,0 +1,77 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+
+const PlansSchema = z
+  .object({
+    owner: z
+      .object({
+        availablePlans: z
+          .array(
+            z
+              .object({
+                baseUnitPrice: z.number(),
+                benefits: z.array(z.string()),
+                billingRate: z.string().nullable(),
+                marketingName: z.string(),
+                monthlyUploadLimit: z.number().nullable(),
+                value: z.string(),
+              })
+              .nullish()
+          )
+          .nullish(),
+      })
+      .nullable(),
+  })
+  .nullable()
+
+const query = `
+  query GetAvailablePlans($owner: String!) {
+    owner(username: $owner) {
+      availablePlans {
+        baseUnitPrice
+        benefits
+        billingRate
+        marketingName
+        monthlyUploadLimit
+        value
+      }
+    }
+  }
+`
+
+export interface UseAvailablePlansArgs {
+  provider: string
+  owner: string
+}
+
+export const useAvailablePlans = ({
+  provider,
+  owner,
+}: UseAvailablePlansArgs) => {
+  // the plans are very static data
```

Comment:
If the plans are very static we could probably set a longer lasting `staleTime` to reduce the amount of requests made overall.